### ISSUE TITLE
FritzBox Cable - Channel Information / Modulation

### ISSUE THREAD

**Issue by leex279** [bb-Ricardo/fritzinfluxdb#100]
As I see now there are no information about the channels/modulation indexed. Would be nice to add this too. 

![image](https://user-images.githubusercontent.com/6664492/225674013-fa3a8c7b-232b-46b7-8241-8116a2b71cc5.png)

Its very useful if you have problems with bad internet quality and can see what is changing and provide information to your provider (currently have this problem).

**Comment by bb-Ricardo** [comment]
Hi,

This could be added. I would need the output of the json response. If you use the browser dev tools you should be able to retrieve them. Also the params which are used to query this data.

**Comment by leex279** [comment]
Hi, 

nice and thanks for the fast reply :) 

I have a litte JS at the moment to extract the upload modulation, if that helps you. Also added the JSON. 
You can find the 2 files here:
https://github.com/leex279/fritzbox-channel/tree/main

Let me know if u need anything else :) 


Thanks

**Comment by bb-Ricardo** [comment]
well, exactly what I needed!

how familiar are you with python?

You could add the data to request here: https://github.com/bb-Ricardo/fritzinfluxdb/blob/main/fritzinfluxdb/classes/fritzbox/service_definitions/connection_info.py

**Comment by leex279** [comment]
I scripted some little things but not very much. But I think I can figure out.

![image](https://user-images.githubusercontent.com/6664492/225835460-ca1d0eef-79e7-4b56-874a-14034a1e39ce.png)

Seems this is working a basic level. I get the values in the log file and can keep go on:
![image](https://user-images.githubusercontent.com/6664492/225835584-2598e1e7-0c46-4b61-92b4-f98b4c5e7a7f.png)

Is there a better way to debug or just like I did with:
./fritzinfluxdb.py -v | grep -A5 -B5 "channelDs"

**Comment by bb-Ricardo** [comment]
That's a good start. I probably have aome time next week to look into it. Which values would be good to monitor? Or just all of them and add them to a dashboard?

**Comment by leex279** [comment]
Sounds good :). I think it would be nice to have all but for my current problem I want to visualize the modulation (type/QAM) and get a stacked bar chart for that, like this:
![image](https://user-images.githubusercontent.com/6664492/226107647-43f7abad-b1bf-4901-8c9d-8f6a98a5ecda.png)

So I can easyli see how often the modulation is changing (according to the vodafone technical guy it should always stay at 64 for all 4 channels :))

**Comment by bb-Ricardo** [comment]
Hi @leex279,

I added the parsing of the data to the "next-release" branch. Can you check out this branch and see if the parsing of the data works?

If it works then only the addition to the system dashboard would be needed.

**Comment by leex279** [comment]
Hi @bb-Ricardo,

yeah, mega cool. Vielen Dank :) Excactly what I need :) 

![image](https://user-images.githubusercontent.com/6664492/226437197-5ef1da50-bc34-4660-a26f-bdf66b417963.png)

**Comment by leex279** [comment]
Also good: State Line:
![image](https://user-images.githubusercontent.com/6664492/226438668-6c0b4e0f-e0c9-4ae5-b0ca-0061d4ef6196.png)

**Comment by bb-Ricardo** [comment]
Sorry for not working on this. It is currently blocked by this issue https://github.com/bb-Ricardo/fritzinfluxdb/issues/114. I need to find a good way forward with influxDB as I don't like to repeat everything once again.

Any ideas from your side?
